### ISSUE TITLE
applies_to_file_regex field in rules is not working

### ISSUE THREAD

**Issue by ismadirolas** [microsoft/ApplicationInspector#408]
**Describe the bug**
If we create a rule with the field 'applies_to_file_regex' it seems to not work properly and we don't get any match.

**To Reproduce**
Create a rule like this:
```
{
  "name": "Test Rule",
  "id": "ID0001",
  "description": "Test Rule",
  "tags":[ "Test.Product.Main" ],
  "severity": "critical",
  "applies_to_file_regex": [ "application.yml" ],
  "patterns": [
    {
      "pattern": "main",
      "type": "regex",
      "scopes": [ "all" ],
      "modifiers": [],
      "confidence": "high"
    }
  ]
}
```
Doesn't detect the word "main" in application.yml file (no matches).

But with applies_to field too:
```
{
  "name": "Test Rule",
  "id": "ID0001",
  "description": "Test Rule",
  "tags":[ "Test.Product.Type.Service" ],
  "severity": "critical",
  "applies_to": [ "yaml" ]
  "applies_to_file_regex": [ "application.yml" ],
  "patterns": [
    {
      "pattern": "main",
      "type": "regex",
      "scopes": [ "all" ],
      "modifiers": [],
      "confidence": "high"
    }
  ]
}
```
Detects the word "main" in application.yml and in other yaml files.

And without the two fields:
```
{
  "name": "Test Rule",
  "id": "ID0001",
  "description": "Test Rule",
  "tags":[ "Test.Product.Type.Service" ],
  "severity": "critical",
  "patterns": [
    {
      "pattern": "main",
      "type": "regex",
      "scopes": [ "all" ],
      "modifiers": [],
      "confidence": "high"
    }
  ]
}
```
Detects "main" in all kind of files.

So, what's happening? Are we using the field in the correct way?

Thanks in advance.

**Comment by gfs** [comment]
You are using it correctly.  There is a bug where we fail to compile the regexes when the Rule is deserialized so the regexes are simply not being run.  Fix is in #409.
